File: examples/dart/basic-bot.dart
import 'dart:io';
import 'package:nyxx/nyxx.dart';
import 'package:dotenv/dotenv.dart';

void main() async {
  final env = DotEnv()..load();
  final token = env['TOKEN']!;
  final prefix = env['PREFIX'] ?? '!';

  final client = await Nyxx.connectGateway(
    token,
    GatewayIntents.allUnprivileged | GatewayIntents.messageContent,
  );

  client.onReady.listen((event) async {
    print('Logged in as ${client.self.username}');

    try {
      await client.http.endpoints.createGlobalSlashCommand(
        client.self.id,
        SlashCommandBuilder('ping', 'Replies with Pong!', []),
      );
      print('Slash command registered');
    } catch (e) {
      print('Failed to register command: $e');
    }
  });

  client.onInteractionCreate.listen((event) async {
    final interaction = event.interaction;
    if (interaction is ISlashCommandInteraction) {
      if (interaction.commandName == 'ping') {
        await interaction.respond(MessageBuilder(content: 'Pong!'));
      }
    }
  });

  client.onMessageCreate.listen((event) async {
    final message = event.message;
    if (message.author.isBot) return;

    final content = message.content;
    if (!content.startsWith(prefix)) return;

    final args = content.substring(prefix.length).split(' ');
    if (args.isEmpty) return;

    if (args.first == 'ping') {
      await message.channel.sendMessage(MessageBuilder(content: 'Pong!'));
    }
  });

  await Future.delayed(Duration.zero);
}

File: examples/kotlin/basic-bot.kt
import dev.kord.core.Kord
import dev.kord.core.event.InteractionCreateEvent
import dev.kord.core.event.MessageCreateEvent
import dev.kord.core.event.ReadyEvent
import dev.kord.core.on
import dev.kord.core.behavior.interaction.respondEphemeral
import dev.kord.common.entity.SlashCommand
import io.github.cdimascio.dotenv.dotenv

suspend fun main() {
  val dotenv = dotenv()
  val token = dotenv["TOKEN"]
  val prefix = dotenv["PREFIX"] ?: "!"

  val kord = Kord(token)

  kord.on<ReadyEvent> {
    println("Logged in as ${kord.self.username}")

    kord.createGlobalChatInputCommand(
      "ping",
      "Replies with Pong!"
    )
    println("Slash command registered")
  }

  kord.on<InteractionCreateEvent> {
    if (interaction.commandName == "ping") {
      interaction.respond { content = "Pong!" }
    }
  }

  kord.on<MessageCreateEvent> {
    if (message.author?.isBot == true) return@on
    val content = message.content
    if (!content.startsWith(prefix)) return@on

    val args = content.removePrefix(prefix).split("\\s+".toRegex())
    if (args.isEmpty()) return@on

    if (args[0] == "ping") {
      message.channel.createMessage("Pong!")
    }
  }

  kord.login()
}
